storage/aptosdb/src/lib.rs

```diff
@@ -1559,6 +1551,23 @@ impl DbWriter for AptosDB {
                     buffered_state.current_state().base_version,
                     buffered_state.current_state().current_version,
                 );
+
+                // Ensure the incoming committing requests are always consecutive.
+                ensure!(
+                    buffered_state.current_state().current_version.map(|version| version + 1).unwrap_or(0) == first_version,
+                    "The first version {} does not follow the current version in buffered state {:?} ",
+                    first_version,
+                    buffered_state.current_state().current_version
+                );
+
+                // Persist ledgerDB data first.
+                {
+                    let _timer = OTHER_TIMERS_SECONDS
+                        .with_label_values(&["save_transactions_commit"])
+                        .start_timer();
+                    self.commit(batch)?;
+                }
```

Comment:
I would (maybe in addition to this) ensure first_version == latset_txn_info_version + 1 from the top of this function